### PR TITLE
Add authorization rules

### PR DESCRIPTION
**Pull request by elmar8287**
- Install CanCanCan in your project.

- Add a role column to the user's table. Remember to use migration for this.

- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.

* For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.

* For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Ability
```
Comment (SajjadAhmad14) [inline app/models/ability.rb:1]: Please use CanCanCan for Post authorization in Post Controller. Please see the example below.
```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi team, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/models/ability.rb (hunk 1 above)
Comment (elmar8287) [reply app/models/ability.rb:1]: Fixed

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🎉🍾✔**
Hello @elmar8287 👋
Your project is complete!  There is nothing else to say other than try and fix your linters after this. it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by elmar8287** [comment]
Thanks for the review! 👍

**Pull request merged**